app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -0,0 +1,445 @@
+            modifier = Modifier.drawOpacity(transitionState[contentAlphaKey])
+        )
+    }
+}
+
+@Composable
+private fun PlantDetailsToolbar(
+    plantName: String,
+    onBackClick: () -> Unit,
+    onShareClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Surface {
+        TopAppBar(
+            modifier = modifier.systemBarsPadding(bottom = false),
+            backgroundColor = MaterialTheme.colors.surface
+        ) {
+            IconButton(onBackClick, Modifier.gravity(Alignment.CenterVertically)) {
+                Icon(Icons.Filled.ArrowBack)
+            }
+            Text(
+                text = plantName,
+                style = MaterialTheme.typography.h6,
+                // As title in TopAppBar has extra inset on the left, need to do this: b/158829169
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxSize()
+                    .wrapContentSize(Alignment.Center)
+            )
+            val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+            IconButton(
+                onShareClick,
+                Modifier.gravity(Alignment.CenterVertically).semantics {
+                    accessibilityLabel = shareAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Share)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImageHeader(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    onFabClick: () -> Unit,
+    isPlanted: Boolean,
+    modifier: Modifier = Modifier,
+    transitionModifier: Modifier = Modifier
+) {
+    var imageHeight by state(StructurallyEqual) { 0 }
+
+    Stack(modifier.fillMaxWidth()) {
+        PlantImage(scrollerPosition, imageUrl, transitionModifier.onPositioned {
+            imageHeight = it.size.height
+        })
+        if (!isPlanted) {
+            val fabModifier = if (imageHeight != 0) {
+                Modifier
+                    .gravity(Alignment.TopEnd)
+                    .padding(end = Dimens.PaddingSmall)
+                    .offset(y = getFabOffset(imageHeight, scrollerPosition))
+                    .plus(transitionModifier)
+            } else {
+                Modifier.visible { false }
+            }
+            val fabAccessibilityLabel = stringResource(R.string.add_plant)
+            FloatingActionButton(
+                onClick = onFabClick,
+                shape = MaterialTheme.shapes.small,
+                modifier = fabModifier.semantics {
+                    accessibilityLabel = fabAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Add)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImage(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    modifier: Modifier = Modifier,
+    placeholderColor: Color = MaterialTheme.colors.onSurface.copy(0.2f)
+) {
+    val parallaxOffset = with(DensityAmbient.current) {
```

Comment:
Can this parallax be implemented using `drawLayer` setting `translationY` rather than `padding` avoiding both layout and round-tripping form Px back to Dps?

Reply:
Tried but doesn't give me the behavior I expect. I indeed want to re-layout the screen as the content moves up and down depending on this